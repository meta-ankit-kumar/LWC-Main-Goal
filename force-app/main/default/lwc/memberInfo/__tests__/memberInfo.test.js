import { createElement } from 'lwc';
import MemberInfo from 'c/memberInfo';
import { props } from './data';

// Helper function to create and set props for the component
function createMemberInfo(props) {
    const element = createElement('c-member-info', {
        is: MemberInfo
    });

    Object.assign(element, props);

    document.body.appendChild(element);

    return element;
}

describe('c-member-info', () => {
    afterEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('should render the component successfully', () => {
        // GIVEN WHEN
        const element = createMemberInfo(props);

        // THEN
        expect(element).toMatchSnapshot();
    });
});
