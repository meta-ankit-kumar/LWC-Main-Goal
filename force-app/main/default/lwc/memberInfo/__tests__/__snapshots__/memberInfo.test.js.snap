// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-member-info should render the component successfully 1`] = `
<c-member-info
  x-test_memberinfo-host=""
>
  #shadow-root(open)
    <div
      class="member-info-container"
      x-test_memberinfo=""
    >
      <p
        x-test_memberinfo=""
      >
        <span
          class="info-type"
          x-test_memberinfo=""
        >
          Name:
        </span>
        <span
          class="info-value"
          x-test_memberinfo=""
        >
          Dummy Member Name
        </span>
      </p>
      <p
        x-test_memberinfo=""
      >
        <span
          class="info-type"
          x-test_memberinfo=""
        >
          Team:
        </span>
        <span
          class="info-value"
          x-test_memberinfo=""
        >
          Dummy Team Name
        </span>
      </p>
      <p
        x-test_memberinfo=""
      >
        <span
          class="info-type"
          x-test_memberinfo=""
        >
          Skills:
        </span>
        <span
          class="info-value"
          x-test_memberinfo=""
        >
          Skills
        </span>
      </p>
    </div>
</c-member-info>
`;
